### PR TITLE
Redesign

### PR DESCRIPTION
**Pull request by lorenzolewis**
- Switch font to Rubik
- Move to using Bootstrap Icons
- Overhaul `<Alert />`
- Overhaul `<Icon />`
- Add light mode
- Enable toggle to switch colour modes
- Update verbiage for announcement bar
- Add light mode mono icon
- Add `theme-color` meta tag to change decorations on supported browser to Tauri Orange
- Remove unneeded `dynamic-css.js`
- Build out semantic colour pallet in CSS
- Add accent colours (WIP and open to modifications to improve contrast)
- Add background gradient animation to each page
- Add hover colour change for links to improve accessibility
- Customise nav bar to make buttons slightly larger
- Fix roadmap line to be lighter and to not go past first and last icon
- Remove unused `#starfield` id in styles
- Remove customised MeiliSearch styling (defaulting to pulling from theme)
- Remove custom overrides used to insert light mode logo previously
- Remove Themify Icons (replaced by Bootstrap Icons)
- Update icons on benchmark and index pages
- Fix code block background colour on light theme
- Update link target size in `<OSList />`
- Reconstruct `SearchBar` component to remove customisations and fallback to theme default
- Add version of Lottie animation on home page for light mode

### CODE REVIEW

**Comment by FabianLars** [comment]
Can we talk about the Roadmap colors again? `Released` vs `To Do` gets a small _yikes_ from me. I mean, they are kinda distinguishable, but i have to ~concentrate to notice the difference (for example this makes them look the same when you're scrolling). Could be worse tho, so i won't complain if we don't find a better alternative.

As always when i'm talking about colors, the disclaimer that i'm colorblind. (Not _completely_ colorblind, that's super duper rare, but "deficiencies" are called blindness too 🙄 )

Edit: also, the hero animation is super laggy as soon as the shadow thingies left and right come into view.

**Comment by lorenzolewis** [comment]
@FabianLars I am very open to discussing the roadmap colours. These are just more of placeholders for the time being to get some feedback

**Comment by amrbashir** [comment]
Good progress so far. A lot has improved but there is still a few things out of place.

- The hamburger menu icon has a lot of margin. Ideally it should have the same margin as the main content. sew below.


-  JS/TS docs states the name of the module 3 times and it feels redundant IMO. I am fine with keeping both 2 & 3 btw.


- Release notes has tow places where I think it could deliver a better experience.
  - on Mobile, packages should be a drop down menu. It will save space and have a better contrast.
  - on Mobile and Desktop, release versions should be also a drop down menu, a lot of space is wasted on mobile because of it, see below.


- Search bar has 3 issues:
  - search icon is always black, and placeholder is always grey. it doesn't adapt to themes which makes it hard to see the icon in the dark theme and hard to see the placeholder in the light theme.
  - search bar styling doesn't match the styling of other input fields on the site, like the release notes filter, see below.

  - search UX is horrible and unusable on mobile, see below. 
Ideally it should be like this on mobie.

**Comment by lorenzolewis** [comment]
@amrbashir I'm going to delay release notes reformatting a bit, tracking in #363 

For the modules page, going to be dependent on changing how we generate docs #364

Search bar just needs a ton of love, but I need to get a re-index of it first

Then for the menu bar padding, I have that on the overall to-do list and tracking https://github.com/tauri-apps/tauri-docs/issues/359

File: src/pages/index.js [hunk 1]
```diff
@@ -1,51 +1,89 @@
...
+
+  return (
+    <lottie-player
+      background="transparent"
+      speed="1"
+      style={{ width: '75%', margin: 'auto' }}
+      autoplay
+    ></lottie-player>
+  )
+}
```
Comment (amrbashir) [inline src/pages/index.js]: Not good at react but I think this is better (well, if it works).

useEffect without a dependency array will run the hook with every single update to the component but we need it only to run when the theme changes. 

```suggestion
  const { isDarkTheme } = useThemeContext()
  const  player = useRef(null)
 
 useEffect(() => {
    player.current.load(
      isDarkTheme ? 'lottie/tauri-splash.json' : 'lottie/tauri-splash-dark.json'
    )
  }, [isDarkTheme])

  return (
    <lottie-player
      ref={player}
      background="transparent"
      speed="1"
      style={{ width: '75%', margin: 'auto' }}
      autoplay
    ></lottie-player>
  )
}
```

**Comment by lorenzolewis** [comment]
@amrbashir How is this for you on mobile around search results? I re-wrote the whole CSS styling and need to adjust colours, decorations, etc., but is the layout working alright for you?

I hard-coded the positioning from the top of the window for now, so I'll work to get that dynamic to rest just below the search bar soon™.

Also haven't touched CSS for the actual text entry field to make it match the rest of the site a bit better.

From what I can tell, the menu button padding seems to be correct now, but let me know if any different for you.
Screenshot 2022-01-23 at 13 49 55
️

**Comment by nothingismagick** [comment]
Can we do this layout? I don't like how the hamburger visually collides with the Tauri logo:

image

**Comment by amrbashir** [comment]
> @amrbashir How is this for you on mobile around search results? I re-wrote the whole CSS styling and need to adjust colours, decorations, etc., but is the layout working alright for you?

Yeah it is working fine, but I still feel strongly that we should mimic the algolia search component (The one I posted above from vue3 docs) only on mobile.

> From what I can tell, the menu button padding seems to be correct now, but let me know if any different for you.

Yeah it is better now.

**Comment by amrbashir** [comment]
> Can we do this layout? I don't like how the hamburger visually collides with the Tauri logo:
> 
> image
> 

I have seen a couple of sites do that nowadays. for example https://deno.land but I perfer the way https://staging.vuejs.org did it.



1. this is the nav bar items. 
2. this is docs sidebar.

**Comment by lorenzolewis** [comment]
> > @amrbashir How is this for you on mobile around search results? I re-wrote the whole CSS styling and need to adjust colours, decorations, etc., but is the layout working alright for you?
> 
> Yeah it is working fine, but I still feel strongly that we should mimic the algolia search component (The one I posted above from vue3 docs) only on mobile.

@amrbashir I agree with you on this, but that's going to be a fairly large undertaking I think, so I'd like to not hold this back for that. There's no documentation or discussions I can find Googling on how to really do this so I'll need to write it from scratch I think.

**Comment by lorenzolewis** [comment]
> Can we do this layout? I don't like how the hamburger visually collides with the Tauri logo:
> 
> image

@nothingismagick so we can do something like this, but I won't be able to move it to the other side without swizzling the component (gets dicy). It has a left container and a right container, and I can't really move it to the other container. I'll open a feature request upstream for this.

Personally, I'd keep it all the way at the edge of the screen as that's the convention. But I do like your suggestion of having it on the other side. 

Screenshot 2022-01-23 at 22 53 24

**Comment by amrbashir** [comment]
> @amrbashir I agree with you on this, but that's going to be a fairly large undertaking I think, so I'd like to not hold this back for that. There's no documentation or discussions I can find Googling on how to really do this so I'll need to write it from scratch I think.

yeah, it can be done later. I will try to help with it once this PR is merged.

**Comment by lorenzolewis** [comment]
FYI, submitted https://docusaurus.io/feature-requests/p/mobile-navbar-toggle-on-righttrailing-edge for trailing edge navbar toggle

**Pull request closed without merging**